tests/stdlib/tstreams4.nim

```diff
@@ -0,0 +1,23 @@
+# bug #9026
+import streams
+
+iterator readFile*(stream: Stream): int =
+  var num = 0
+  var line = ""
+  while stream.readLine(line):
+    num += 1
+    yield line.len
+
+iterator readFile*(s: string): int =
+  for x in readFile(newStringStream(s)):
+    yield x
+
+let
+  filenames = ["bin.txt", "str.txt"]
```

Comment:
* instead of committing 2MB of random binary data to nim repo (it's not insignificant, really), why not simply auto-generate (with a fixed seed) random chars and assert that does what we expect? (and remove the file upon test completion)

* that function that generates text data could also be useful once we fix the 100x speed issue (https://github.com/nim-lang/Nim/issues/9975)